### DISCUSSION TITLE
How to set policy to enable the kernel to encrypt two layers of ESP on data?

### DISCUSSION THREAD

**Discussion by Kev-Jiang** [strongswan/strongswan#2174] [Q&A]
I hope the data will pass through the first policy(reqid 9 mode transport) encryption, and then the second policy(reqid 1 mode tunnel) encryption.But now the kernel will only execute the first policy.
So, how to set policy to enable the kernel to encrypt two layers of ESP on data?

The current policyis as follows：
src 2a00:23ee:f178:5eee:1e41:c2dc:40f2:ea4b/128 dst 2a01:4c8:f000:f54::1/128 sport 8180 dport 5063 
	dir out priority 268799 
	tmpl src :: dst ::
	proto esp spi 0x19ef536f reqid 9 mode transport
src 2a00:23ee:f178:5eee:1e41:c2dc:40f2:ea4b/128 dst 2a01:4c8:f000::/48 
	dir out priority 309887 
	tmpl src 192.168.1.103 dst 46.68.59.19
	proto esp spi 0xd1046e6a reqid 1 mode tunnel

**Comment by tobiasbrunner** [comment]
I think this is possible via XFRM interfaces (route ESP packets from the first SA through an XFRM interface that applies the second). Not sure if it's possible without (maybe with the "set in/out mark" option).

**Reply by Kev-Jiang** [reply]
Can it be achieved using strongswan? Can you help provide a demo？

**Reply by tobiasbrunner** [reply]
You might want to refer to [the docs](https://docs.strongswan.org/docs/5.9/features/routeBasedVpn.html#_xfrm_interfaces_on_linux).

**Reply by Kev-Jiang** [reply]
Thank you very much. I will review this document again.
